src/full_node/block_store.py

```diff
@@ -138,6 +138,28 @@ async def get_sub_block_records(
                 peak = header_hash
         return ret, peak
 
+    async def get_sub_block_in_range(
```

Comment:
Note that this might return an unbounded number of sub-blocks, since there can be more than one per height. Why don't we backtrack from a specific header hash instead (at the stop index), so that we add the minimum number of blocks to the cache?

Reply:
i think that traverse means db access's on each record,

why do we persist records that dont belong to "correct" chain ? 
just in case we crash in sync ?
if so when do we clean that up ?